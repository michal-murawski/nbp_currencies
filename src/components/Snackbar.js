import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Button from 'material-ui/Button';
import SnackbarMUI from 'material-ui/Snackbar';
import { errorsClearMessage as errorsClearMessageAction } from 'store/errors/actions';
import { getValueByPath } from 'utils/dataHelpers';

class Snackbar extends React.PureComponent {
  handleOnClick = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    this.props.errorsClearMessage();
  };

  render() {
    const { message } = this.props;

    return (
      <SnackbarMUI
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        open={Boolean(message)}
        autoHideDuration={6000}
        onClose={this.handleOnClick}
        SnackbarContentProps={{
          'aria-describedby': 'message-id',
        }}
        message={message}
        action={[
          <Button key="close" color="secondary" size="small" onClick={this.handleOnClick}>
            Close
          </Button>,
        ]}
      />
    );
  }
}

Snackbar.propTypes = {
  message: PropTypes.string,
  errorsClearMessage: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  message: getValueByPath(['errors', 'message'], state),
});

const mapDispatchToProps = dispatch => ({
  errorsClearMessage: () => dispatch(errorsClearMessageAction()),
});

export const SnackbarRaw = Snackbar;

export default connect(mapStateToProps, mapDispatchToProps)(Snackbar);
