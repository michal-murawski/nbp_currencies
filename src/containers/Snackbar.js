import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Button from 'material-ui/Button';
import SnackbarMUI from 'material-ui/Snackbar';
import { errorsClearMessage as errorsClearMessageAction } from 'store/errors/actions';
import { getValueByPath } from 'utils/dataHelpers';

function Snackbar({ message, errorsClearMessage }) {
  return (
    <SnackbarMUI
      anchorOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
      open={Boolean(message)}
      autoHideDuration={6000}
      onClose={errorsClearMessage}
      SnackbarContentProps={{
        'aria-describedby': 'message-id',
      }}
      message={message}
      action={[
        <Button
          key="close"
          color="secondary"
          size="small"
          onClick={errorsClearMessage}
        >
          Close
        </Button>
      ]}
    />
  );
}

Snackbar.propTypes = {
  message: PropTypes.string,
};

const mapStateToProsp = ({
  message: getValueByPath(['errors', 'message']),
});

const mapDispatchToProsp = (dispatch) => ({
  errorsClearMessage: () => dispatch(errorsClearMessageAction()),
});


export default connect(mapStateToProsp, mapDispatchToProsp)(Snackbar);
