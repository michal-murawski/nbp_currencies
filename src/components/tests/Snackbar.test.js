import React from 'react';
import { shallow } from 'enzyme';
import SnackbarMUI from 'material-ui/Snackbar';
import toJson from 'enzyme-to-json';

import { SnackbarRaw } from '../Snackbar';

const mockClearMessage = jest.fn();

describe('components/SnackbarRaw', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SnackbarRaw errorsClearMessage={mockClearMessage} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should be closed without message', () => {
    expect(wrapper.find(SnackbarMUI).props().open).toBe(false);
  });

  it('should be get open after passing message', () => {
    const message = 'Test message';
    expect(wrapper.find(SnackbarMUI).props().open).toBe(false);

    wrapper.setProps({ message });

    expect(wrapper.find(SnackbarMUI).props().open).toBe(true);
    expect(wrapper.find(SnackbarMUI).props().message).toEqual(message);
    expect(wrapper.props().message).toEqual(message);
  });

  it('should call proper function on button click', () => {
    expect(mockClearMessage).toHaveBeenCalledTimes(0);

    const snackbarCloseButton = wrapper.find(SnackbarMUI).props().action[0];
    shallow(snackbarCloseButton).simulate('click');

    expect(mockClearMessage).toHaveBeenCalledTimes(1);
  });
});
