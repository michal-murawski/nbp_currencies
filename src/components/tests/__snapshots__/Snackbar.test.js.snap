// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SnackbarRaw should render properly 1`] = `
<WithStyles(Snackbar)
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  action={
    Array [
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        size="small"
    >
        Close
    </WithStyles(Button)>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={6000}
  onClose={[Function]}
  open={false}
/>
`;
